File: java/std_nat/src/fan/std/FilePeer.java
package fan.std;

import fan.sys.IOErr;
import fan.sys.List;

public class FilePeer {
	static File make(Uri uri, boolean checkSlash) {
		String path = uri.pathStr;
		java.io.File jfile = new java.io.File(path);
	    if (jfile.isDirectory() && !checkSlash && !uri.isDir())
	      uri = uri.plusSlash();
	    return LocalFilePeer.make(jfile, uri);
	}
	
	static File make(Uri uri) {
		return make(uri, true);
	}
	
	static File os(String osPath) {
		return LocalFilePeer.make(osPath);
	}
	
	static List osRoots() {
		java.io.File[] roots = java.io.File.listRoots();
		List list = List.make(roots.length);
	    for (int i=0; i<roots.length; ++i) {
	      list.add(LocalFilePeer.make(roots[i]));
	    }
	    return list;
	}
	
	static File createTemp(String prefix, String suffix, File dir) {
		if (prefix == null || prefix.length() == 0) prefix = "fan";
	    if (prefix.length() == 1) prefix = prefix + "xx";
	    if (prefix.length() == 2) prefix = prefix + "x";

	    if (suffix == null) suffix = ".tmp";

	    java.io.File d = null;
	    if (dir != null)
	    {
	      if (!(dir instanceof LocalFile)) throw IOErr.make("Dir is not on local file system: " + dir);
	      d = LocalFilePeer.getJFile((LocalFile)dir);
	    }

	    try
	    {
	      return LocalFilePeer.make(java.io.File.createTempFile(prefix, suffix, d));
	    }
	    catch (java.io.IOException e)
	    {
	      throw IOErr.make(e);
	    }
	}
	
	static String sep() { return java.io.File.separator; }
	
	static String pathSep() { return java.io.File.pathSeparator; }
}

File: src/std/js/FilePeer.js
fan.std.FilePeer = function(){}


fan.std.FilePeer.make = function(uri, checkSlash) {
	if (checkSlash == undefined) checkSlash = true;
	self = fan.std.File.make();
	self.m_file = new File("", uri.toStr());
	return self;
}

fan.std.FilePeer.os = function(osPath) {
	self = fan.std.File.make();
	self.m_file = new File("", osPath);
	return self;
}

fan.std.FilePeer.osRoots = function(osPath) {
	return new fan.sys.List.make();
}

fan.std.FilePeer.createTemp = function(prefix, suffix, dir) {
	if (prefix == null || prefix.length() == 0)
		prefix = "fan";
	if (prefix.length() == 1)
		prefix = prefix + "xx";
	if (prefix.length() == 2)
		prefix = prefix + "x";

	if (suffix == null)
		suffix = ".tmp";

	if (dir != null) {
		return dir + suffix;
	}

	return fan.std.FilePeer.os("/temp/"+suffix);
}

fan.std.FilePeer.sep = function() {
	return "/";
}


fan.std.FilePeer.pathSep = function() {
	return ":";
}

fan.std.FilePeer.plusNameOf = function(self, x) {
	var name = x.name();
	if (x.isDir())
		name += "/";
	return self.plus(Uri.fromStr(name));
}

fan.std.FilePeer.copyTo = function(self, to, options) {
	
}
